controllers/vaultdynamicsecret_controller.go

```diff
@@ -288,3 +304,55 @@ func isLeaseNotfoundError(err error) bool {
 	}
 	return false
 }
+
+// handleDeletion will handle the deletion path of the VDS secret:
+// * revoking any associated outstanding leases
+// * removing our finalizer
+func (r *VaultDynamicSecretReconciler) handleDeletion(ctx context.Context, o *secretsv1alpha1.VaultDynamicSecret) error {
+	logger := log.FromContext(ctx)
+	if err := r.revokeLease(ctx, o, ""); err != nil {
+		// Do not return error, otherwise we may fail to remove the finalizer.
+		// Worst case at this point we will leave a dangling lease instead of a secret which
+		// cannot be deleted.
+	}
```

Comment:
The comment is great, but should we log something here? Otherwise the `if err := ...` seems extraneous (and go-staticcheck is complaining to me about it 🙂).

Reply:
Good call, we're actually emitting an Event and logging inside this function in those cases, I'm going to change the function signature a bit so we're not "hiding anything" and move the comment inside `revokeLease`